### PR TITLE
Add Odoo 17.0

### PR DESCRIPTION
**Pull request by pedrobaeza**
- Dockerfile copied from 16.0 and slightly adapted
- Fix tests
- Make it `:latest`.
- Deprecate some old Odoo versions.
- Bump PG to 15
- Switch to Python 3.10
- Adapted CI

@Tecnativa TT45808

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@yajo @celm1990 I'm not getting this to upload the `-onbuild` image, and that it's making the CI to fail:

```
ERROR: failed to solve: tecnativa/doodba:17.0-onbuild: docker.io/tecnativa/doodba:17.0-onbuild: not found
```

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -35,23 +35,26 @@ jobs:
...
+            odoo_version: "14.0"
+            python_version: "3.9"
+          - pg_version: "14"
             odoo_version: "13.0"
+            python_version: "3.9"
           - pg_version: "13"
-            odoo_version: "12.0"
-          - pg_version: "12"
-            odoo_version: "11.0"
+            odoo_version: "13.0"
```
Comment (yajo) [inline .github/workflows/ci.yaml:53]: **thought:** I find it weird that you test odoo 13 twice and you don't run odoo 11, 12 or 16.

Other than that, I see no reason why builds for 17 get aborted.

Comment (pedrobaeza) [reply .github/workflows/ci.yaml:53]: Ouch, my fault! I want it to test it just once for 17, 16, 15, 14 and 13, and remove official support for 11 and 12.

**Comment by celm1990** [comment]
I think this log in GitHub Actions might be related. It was showing intermittent issues a few days ago. I'm not sure how much disk space GitHub provides, but could it be that the tests generate a lot of logs?

image

**Pull request merged**